stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -245,11 +246,6 @@ private static MethodSpec getReadMethodSpec(@NotNull TypeName typeName,
                         "();\n" +
                         "\twhile (reader.hasNext()) {\n" +
                         "\t\tString name = reader.nextName();\n" +
-                        "\t\tcom.google.gson.stream.JsonToken jsonToken = reader.peek();\n" +
-                        "\t\tif (jsonToken == com.google.gson.stream.JsonToken.NULL) {\n" +
-                        "\t\t\treader.skipValue();\n" +
-                        "\t\t\tcontinue;\n" +
-                        "\t\t}\n" +
```

Comment:
I'm interested in what the rational is for this besides performance improvement. I see it as a potential problem because we have two potential flows for primitives (objects would be fine because we would resort to the object type adapters which check for null).
1. Parsed via a type adapter that is in the gson `TypeAdapters` class: e.g. booleans are parsed using `TypeAdapters.BOOLEAN`. This adapter does a null check and returns null of the token is null. We then do a null check and optionally assign the field depending on if the item returned was null. This means the field remains uninitialized.
2. Parsed via a type adapter that is in `KnownTypeAdapters`. This adapter does not do a null check and will throw an illegal state exception because the peeked type did not match the type that it wanted (since the fields are ultimately assigned via `reader.nextType`.

Am I misunderstanding something? I think we should be consistent for all primitives